pkg/client/factory_test.go

```diff
@@ -201,3 +202,17 @@ func TestClientWithDatafileAccessToken(t *testing.T) {
 
 	assert.Equal(t, accessToken, factory.DatafileAccessToken)
 }
+
+func TestClientWithDefaultDecideOptions(t *testing.T) {
```

Comment:
add same case, u modify decideOptions outside, that should not impact `optimizelyClient.defaultDecideOptions`